### PR TITLE
✨  Aave Proposal Pivotality Report

### PR DESCRIPTION
**Pull request by acemasterjb**
See https://github.com/acemasterjb/gov_analysis/pull/6

### CODE REVIEW

File: book/aave_proposals.ipynb [hunk 1]
```diff
@@ -0,0 +1,1182 @@
+{
```
Comment (lajarre) [inline book/aave_proposals.ipynb:1]: Notebooks are getting updated via the Github action only on changes to `main`.

Hence, it's important to show to the reader the date at which the notebook has been run, to know how fresh is the data.

I suggest we add the date right below the title of all notebooks.

File: .github/workflows/publish_report_aave.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+name: publish-book-aave
```
Comment (lajarre) [inline .github/workflows/publish_report_aave.yml:1]: We must no change the contents of the plutocracy report book, published at https://butterymoney.github.io/gov_analysis/report/plutocracy_report.html

We must keep this report final, as it's already been published together with a Mirror article.

So, I suggest deleting the `publish_report.yml` file or commenting-out its contents.

File: book/aave_proposals.ipynb [hunk 3]
```diff
@@ -0,0 +1,1182 @@
...
+ "cells": [
```
Comment (lajarre) [inline book/aave_proposals.ipynb:2]: My general comment on the contents of the report, cc @vaughnmck for further review (you can look at the rendered Jupyter notebook by clicking on View File 
image
).

> We take a look at 5 recent proposals on Aave Governance, including the first Butter Aave Elections, and compare the voter turnout and the "pivotality" (a measure of whale influence) of each of them.

This is not what we planned. We want to take a look at the top 10 Aave proposals on Snapshot ranked by Voting Power and #Voter. Here there are, based on @vaughnmck 's analysis:

<google-sheets-html-origin><style type="text/css"></style>

Title | Total VP | Voters | Total VP Rank | Voter Rank | Total Rank | ID
-- | -- | -- | -- | -- | -- | --
[TEMP CHECK] Incentivized Delegate Campaign (3-month) | 742,542.48 | 17,658.00 | 6.00 | 3.00 | 9.00 | 0xf6da417983653f7e7f5560a8e16ad77d5e25533bba9157abfd222347b40d09d0
[TEMP CHECK] - Launch Aave V3 on Starknet | 665,006.49 | 21,321.00 | 16.00 | 1.00 | 17.00 | 0xde90f4d29c4c5dd55cba023b29bdde8c49e37f70c3398668a78e2f173a61b4e6
Safety Module upgrade to v1.5 | 728,994.69 | 16,152.00 | 7.00 | 20.00 | 27.00 | 0x743ae6001bb0fdfa9a8c4475a2489c20917a79084725f0101ac7c1e7f0cf27c6
Deploy Aave V3 to Metis Andromeda Mainnet | 717,974.83 | 16,126.00 | 9.00 | 22.00 | 31.00 | 0x7e6175c21342dbd88dc5b0dabf2d0a86339e3ef3f199fc7d778a9a6f05ed0df9
[ARFC] Add MKR to Ethereum v3 | 660,111.69 | 16,354.00 | 17.00 | 17.00 | 34.00 | 0xf4aec3fbab5096752be96f0e5b522f37318c1902cf8b897b049b7a94d478de73
[TEMP CHECK] Aave V3 deployment on BNB Chain | 586,055.57 | 16,976.00 | 32.00 | 6.00 | 38.00 | 0x4579b562a4ea07c52871f724a8baf156514a97384e9427156d04fb6f0ae91515
[ARFC] Grant OP Emission_Admin for Optimism v3 Liquidity Pool to Lido DAO | 579,100.80 | 17,663.00 | 38.00 | 2.00 | 40.00 | 0xd8a400fc731b6e807f4d89aa1fa47225f0dfaaa8fc49ef04b46b11b98fbfa15f
[ARFC] Add UNI to Ethereum v3 | 618,327.88 | 16,482.00 | 28.00 | 12.00 | 40.00 | 0x51d67ef69e901b34f1d111f2cd5d582c59cffa8d70b7939023febd20f7613b88
[ARFC] Deploy aCRV & CRV to veCRV | 585,810.01 | 16,462.00 | 33.00 | 13.00 | 46.00 | 0xf92c5647c7f60a4a3db994b4953fc4408f5946cafdc0cebcd4c5924f40e04d36
[ARFC] Add SNX to Ethereum v3 | 619,021.56 | 16,240.00 | 27.00 | 19.00 | 46.00 | 0x5f232a89e10d67df3aad2907e8dce3bec9708596929b3254055cf37499969b89

Can you please update the analysis with this?

## Below my wording comments

> We take a look at 5 recent proposals on Aave Governance, including the first Butter Aave Elections, and compare the voter turnout and the "pivotality" (a measure of whale influence) of each of them.

replace with

> We take a look at the top 10 Aave Snapshot proposals ranked by voting power and voters count, includes the first Butter Aave Elections.
> 
> We compare their voter turnout and _pivotality_ metric (a measure of whale influence, described in our previous [Mirror article](https://mirror.xyz/butterd.eth/_VrHsBXZU0IYuIUMq0ifJTffvH0e1AKNh6I-BIV3Nc4) and [data report](https://butterymoney.github.io/gov_analysis/report/plutocracy_report.html)).

----

> ## Case Studies

replace with

> ## Comparison of the top 10 proposals

----

> ### Proposal Analysis

replace with

> ## Case studies

----

Keep only a line separator (you know, the visual horizontal lines) right before case studies, and remove other ones.

**Review by lajarre — CHANGES_REQUESTED** [review]
Please review in-line comments

File: book/aave_proposals.ipynb (hunk 3 above)
Comment (acemasterjb) [reply book/aave_proposals.ipynb:2]: Thanks for your review!

Comment (acemasterjb) [reply book/aave_proposals.ipynb:2]: > Here there are, based on @vaughnmck 's analysis

The top 10 listing would be different from yours, right?

This is what I get when sorting that column in his dataset

image


My sorting is slightly different

image

I've noticed this for a while, the difference is due to the amount of data I'm getting; the oldest proposal I'm getting ended in 08/12/22, the first Gauntlet <> Aave renewal proposal for example ended in 23/11/22.

I'll try running again w/ last 150 proposals to see if I get a similar list for top 10 proposals to @vaughnmck's.

**Comment by acemasterjb** [comment]
Made a note about the sample size for the "proportion of outcomes changed (%)" stat.

> 9.33% of Aave's proposal outcomes (of the 150 sampled) change after filtering out whale voting power.

**Comment by acemasterjb** [comment]
Still need clarification on the ordering of proposals, and data source for rankings.

**Comment by lajarre** [comment]
> Still need clarification on the ordering of proposals, and data source for rankings.

Here is the ranking you should make use of:



1 | [TEMP CHECK] Incentivized Delegate Campaign (3-month) | 0xf6da417983653f7e7f5560a8e16ad77d5e25533bba9157abfd222347b40d09d0 | 17658 | 742542.4820297458
-- | -- | -- | -- | --
2 | [TEMP CHECK] - Launch Aave V3 on Starknet | 0xde90f4d29c4c5dd55cba023b29bdde8c49e37f70c3398668a78e2f173a61b4e6 | 21321 | 665006.4912716972
3 | [ARC] E-Mode Parameter Changes for V3 Avalanche, Optimism, Polygon, and Arbitrum | 0x84deca82139320b2570f04211b249e37b8a7602b4a0ed70e6fa772c9f6e94550 | 18710 | 659590.8795181853
4 | [TEMP CHECK] - Whitelist Stargate for V3 Portals | 0xc17bc863a09c72152c1ac25d09fcee4748b58eb2e10e70886e20e77ad4d8fad0 | 18758 | 654740.4896129176
5 | Safety Module upgrade to v1.5 | 0x743ae6001bb0fdfa9a8c4475a2489c20917a79084725f0101ac7c1e7f0cf27c6 | 16152 | 728994.6873724599
6 | Deploy Aave V3 to Metis Andromeda Mainnet | 0x7e6175c21342dbd88dc5b0dabf2d0a86339e3ef3f199fc7d778a9a6f05ed0df9 | 16126 | 717974.8308977766
7 | [ARFC] Add MKR to Ethereum v3 | 0xf4aec3fbab5096752be96f0e5b522f37318c1902cf8b897b049b7a94d478de73 | 16354 | 660111.6887886972
8 | [TEMP CHECK] Aave V3 Deployment on zkSync Era Mainnet | 0x46cf72da892eb216edc1b7fe2f24f7491d8c37344b2b1f67632fa6950be034f7 | 18546 | 579204.9965656479
9 | [TEMP CHECK] Aave V3 deployment on BNB Chain | 0x4579b562a4ea07c52871f724a8baf156514a97384e9427156d04fb6f0ae91515 | 16976 | 586055.5706328462
10 | [ARFC] Grant OP Emission_Admin for Optimism v3 Liquidity Pool to Lido DAO | 0xd8a400fc731b6e807f4d89aa1fa47225f0dfaaa8fc49ef04b46b11b98fbfa15f | 17663 | 579100.7982572629



This ranking is taken from the following query: https://dune.com/autopoiesis/aave-top-10-snapshot-proposals

The ordering function used is pseudocoded `rank(voting power) + rank(voter count)`.

File: .github/workflows/publish_report_aave.yml (hunk 2 above)
Comment (lajarre) [reply .github/workflows/publish_report_aave.yml:1]: @acemasterjb is the branch up to date? I'm still seeing the `publish_report.yml` file

File: book/aave_proposals.ipynb [hunk 4]
```diff
@@ -0,0 +1,1689 @@
+{
```
Comment (lajarre) [inline book/aave_proposals.ipynb:1]: Some further fixes:

[] Rename title to `Whale Pivotality in Aave top Snapshot proposals`

[] Move the `<hr/>` just above Case Study

image

**Review by lajarre — CHANGES_REQUESTED** [review]

**Comment by acemasterjb** [comment]
> > Still need clarification on the ordering of proposals, and data source for rankings.
> 
> Here is the ranking you should make use of:
> This ranking is taken from the following query: https://dune.com/autopoiesis/aave-top-10-snapshot-proposals
> 
> The ordering function used is pseudocoded `rank(voting power) + rank(voter count)`.

As I'm working on this, I've encountered an issue with providing this ordering. [I can't request more than 6000 entities on Snapshot's GraphQL API](https://hub.snapshot.org/graphql?query=query%20Votes%20%7B%0A%20%20votes%20(%0A%20%20%20%20first%3A%201000%0A%20%20%20%20skip%3A%206000%0A%20%20%20%20where%3A%20%7B%0A%20%20%20%20%20%20proposal%3A%20%220x037a5ef4ac471dfc07054c01f909b9a71d7bf5ffdfe9b959121bf60d600427b4%22%0A%20%20%20%20%7D%0A%20%20%20%20orderBy%3A%20%22vp%22%2C%0A%20%20%20%20orderDirection%3A%20desc%0A%20%20)%20%7B%0A%20%20%20%20id%0A%20%20%20%20voter%0A%20%20%20%20created%0A%20%20%20%20proposal%20%7B%0A%20%20%20%20%20%20id%0A%20%20%20%20%20%20scores%0A%20%20%20%20%20%20scores_state%0A%20%20%20%20%7D%0A%20%20%20%20choice%0A%20%20%20%20vp%0A%20%20%20%20metadata%0A%20%20%20%20space%20%7B%0A%20%20%20%20%20%20id%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D%0A).

~~Right now, I think one solution could be to use the Dune API to query the referenced query and grab the top 10 `proposal_id`s from there. Then I can just make a dataframe mask and get the rows for those top 10 proposals.~~

### Edit

Solution: Hardcode the `proposal_id`s from the [Dune query](https://dune.com/autopoiesis/aave-top-10-snapshot-proposals), make a mask using it, apply it to the proposal DFs and preserve the ordering from source (Dune query).

File: book/aave_proposals.ipynb (hunk 4 above)
Comment (acemasterjb) [reply book/aave_proposals.ipynb:1]: On your first remark:

This workflow file is [removed in the latest commit](https://github.com/acemasterjb/gov_analysis/tree/617dc74321f02d0137408d277707aedbbf7e140a/.github/workflows) from upstream. It should not be there. Please confirm.

**Review by lajarre — APPROVED** [review]
lgtm

**Pull request merged**